sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp

```diff
@@ -576,6 +576,7 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
       mutable Credentials::AccessToken m_accessToken;
       mutable std::shared_timed_mutex m_accessTokenMutex;
       mutable Credentials::TokenRequestContext m_accessTokenContext;
+      mutable std::atomic<bool> m_invalidateToken = {false};
```

Comment:
I don't understand why this needs to be `std::atomic<bool>`. In all the processors that we support, reads and writes to POD data is atomic. For types which support modification (`++`, `+=`, etc.) there are strong reasons for using `std::atomic`, but for boolean, I'm not sure that there are strong reasons.

I believe that the `m_invalidateToken` variable and the cached token should both be protected by the `m_accessTokenMutex`, which alleviates any requirement that the `m_invalidateToken` variable be atomic.

The use of `std::atomic` isn't automatically a code smell, but it does require additional scrutiny (because the use of `std::atomic` is an indication that the developer is attempting to write lock-free code, and that is an arena where developers should fear to tread.

Reply:
I initially had it as a bool (and agree with your point about bools having atomic read/writes).
I changed it to atomic based on Anton's feedback.

I will revert that. I don't believe setting the bool requires a mutex here (and modifying the cached token is already under one).

Reply:
@LarryOsterman, over there it is not only for atomicity, but for concurrent-safety as well, so that the compiler does not optimize or rearrange reads. Better to be safe there.
`m_invalidateToken` is not protected by `m_accessTokenMutex` on the instance of setting it to false. And using a unique lock to set to to false, would be an overkill, especially that just some time later it will go into a function to once again obtain a unique lock to actually refresh the token.
It does not cost us anything to keep the atomic bool there.

Reply:
Okay, keeping the atomic then. :p